consutil/lib.py

```diff
@@ -327,14 +330,23 @@ def _parse_processes_info(output):
     @staticmethod
     def run_command(cmd, abort=True):
         """runs command, exit if stderr is written to and abort argument is ture, returns stdout, stderr otherwise"""
-        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, text=True)
+        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
         output = proc.stdout.read()
         error = proc.stderr.read()
         if abort and error != "":
             click.echo("Command resulted in error: {}".format(error))
             sys.exit(ERR_CMD)
         return output if abort else (output, error)
 
+    @staticmethod
+    def run_command_pipe(*args, abort=True):
+        """runs command, exit if stderr is written to and abort argument is ture, returns stdout, stderr otherwise"""
+        exitcodes, output = getstatusoutput_noshell_pipe(*args)
+        if abort and any(exitcodes) and output != '':
```

Comment:
>output [](http://example.com/codeflow?start=40&length=6)

The original function `run_command` is checking stderr, but you are checking stdout. Did you change the behavior?

Reply:
in case `getstatusoutput_noshell_pipe` has errors or not, it both writes to stdout